articles/libraries/lock-android/native-provider-google.md

```diff
@@ -107,82 +116,62 @@ private void beginAuthentication(){
+### Requesting a custom Google scope
 
-### Log out / Clear account.
-To log out the user so that the next time they are prompted to input their credentials call `clearSession`. After you do this the provider state will be invalid and you will need to call `start` again before trying to `authorize` a result. Calling `stop` has the same effect.
+By default, the scope `Scopes.PLUS_LOGIN` is requested. You can customize the Scopes by calling `setScopes` with the list of Scopes. Each Google API (Auth, Drive, Plus..) specify it's own list of Google Scopes.
 
 ```java
-provider.clearSession();
+provider.setScopes(Arrays.asList(new Scope(Scopes.PLUS_ME), new Scope(Scopes.PLUS_LOGIN)));
 ```
 
-### Remember the Last Login
-By default, this provider will remember the last account used to log in. If you want to change this behavior, use the following method.
+### Requesting custom Android runtime permissions
+
+This provider doesn't require any special _Android Manifest Permissions_ to authenticate the user. But if your use case requires them, you can let the AuthProvider handle them for you. Use the `setRequiredPermissions` method.
 
 ```java
-provider.rememberLastLogin(false);
+provider.setRequiredPermissions(new String[]{"android.permission.GET_ACCOUNTS"});
 ```
 
-### Certificate Fingerprints
-When creating a new OAuth Credential in the Google Developers Console you will need to provide the SHA-1 of the certificate you're using to sign your application. When completing your Client's Configuration in the Auth0 Dashboard you will also need to provide the SHA-256 value. To obtain them follow this steps:
+If you're not using Lock then you'll have to handle the permission request result yourself. To do so, make your activity implement the `ActivityCompat.OnRequestPermissionsResultCallback` interface. When the `onRequestPermissionsResult` method gets called pass the result to the provider by calling `provider.onRequestPermissionsResult`.
 
-Locate the certificate you're using to sign your application. If you don't have one you can generate it or use the default `android.keystore` certificate, that was generated automatically for you when you installed the SDK. In this example we're going to use the default one.
+### Log out / clear account.
 
-* On Windows:
+To log out the user so that the next time they are prompted to input their credentials call `clearSession`. After you do this the provider state will be invalid and you will need to call `start` again before trying to `authorize` a result. Calling `stop` has the same effect.
 
-```sh
-keytool -list -v -keystore "%USERPROFILE%\.android\debug.keystore" -alias androiddebugkey -storepass android -keypass android
+```java
+provider.clearSession();
 ```
 
-* On Linux / macOS
+### Remember the last login
 
-```sh
-keytool -list -v -keystore ~/.android/debug.keystore -alias androiddebugkey -storepass android -keypass android
```

Comment:
Keep this, maybe we can avoid the windows/mac diff snippet by just saying "path-to-the-keystore"